### ISSUE TITLE
How to configure Energie Dashboards (after SENEC.Home have been updated to 0825+)

### ISSUE THREAD

**Issue by marq24** [marq24/ha-senec-v3#11]
As explained in #4 after your SENEC.Home System has been updated (remotely) to **REVISION MCU 0825** by SENEC (without asking) the previously available statistics data is not available any longer in the default integration.

As also described in #4  you have the option to install an additional Integration Entry **WEB.API: mein-senec.de Portal (usable with all SENEC.Home variants)** - you need to update the integration `3.0.6` or higher (any `20YY.MM.N` Version will work as well)

So then you are facing the challenge, that you have old sensor data (before the 0825 update) and the new sensor data that you might like to display together...

My personal current solution is to edit the Energy Dashboard in the following way:
![image](https://github.com/marq24/ha-senec-v3/assets/8012359/94351a49-b801-42a7-8e61-eada7944fddd)

and **add** for all the four fields the new sensors - so it looks like this (please note, that I have used `senecw` as my WebAPI Integration name (the default is `senec_WEBAPI`) - after you edited the dashboard configuration it could look like this:
![image](https://github.com/marq24/ha-senec-v3/assets/8012359/90232f56-c735-4444-8f9c-03e5b6f4d2e9)

Now you see on the Energy Dashboard your "old" and "new" data together... (yes in slightly different colour shades)

So yes - the old sensors are still reported as "missing" - but IMHO this is a valid compromise - if you have alternative ideas (or opinions) - please feel free to share!

## Update - for advanced Users:
@loeffelpan have provided information how you can manipulate HA's SQLite-DB (https://github.com/marq24/ha-senec-v3/discussions/17).

In the initial post you will find [a link](https://gist.github.com/loeffelpan/10bdb7e5bfd21568a9cb05d429aaf214) to an initial Windows PowerShell script (that you must adjust to your local setup/needs) that allows you to add energy data from the past to your energy dashboard.
Additional there are instructions how to 'join' energy data from old and new sensors (https://github.com/marq24/ha-senec-v3/discussions/17#discussioncomment-7094001).

**Comment by Kuechenfenster** [comment]
danke, looks great.
one thing, die import and export Euro/kwH werde fuer die "veralteten" daten nicht mehr uebernommen..

<img width="1078" alt="Screenshot 2023-09-08 at 15 08 39" src="https://github.com/marq24/ha-senec-v3/assets/73640595/f568d953-3b98-4d49-a2c8-d4f740b64591">

**Comment by mschmiedel** [comment]
Puh, this is quite interesting. I have been using the integration that this one is forked from for quite a while and did some manual changes to get Wallbox working and other stuff. Switch to this integration was easy,

However, I have been using the integral helpers to get power consumption, however these have some minor precision issues (sample every 30s means peaks under 30s are not counted). I will at some point switch to the web api data. Has anyone here done the migration of the data in the db and has a base script for this?

**Comment by marq24** [comment]
> However, I have been using the integral helpers to get power consumption, however these have some minor precision issues (sample every 30s means peaks under 30s are not counted). I will at some point switch to the web api data. Has anyone here done the migration of the data in the db and has a base script for this?

Would this help?
https://github.com/marq24/ha-senec-v3/discussions/17

**Comment by mschmiedel** [comment]
> > However, I have been using the integral helpers to get power consumption, however these have some minor precision issues (sample every 30s means peaks under 30s are not counted). I will at some point switch to the web api data. Has anyone here done the migration of the data in the db and has a base script for this?
> 
> Would this help? #17

Yes, thank you. I am though wondering if it makes sense to use the online data if it is not clearly better / more precise than the results of the integral functions... Any thoughts about that?

**Comment by marq24** [comment]
> Any thoughts about that?

nothing that based on facts - just 100% pure opinion - I use here the web-api - but only cause this helps me to keep track of possible API changes/errors that SENEC introduce over time. If you feel comfortable with the local integral based solution then I would keep that (since everything is in your control)
